src/Storages/WindowView/StorageWindowView.cpp

```diff
@@ -1449,6 +1449,8 @@ void StorageWindowView::checkTableCanBeDropped() const
 
 void StorageWindowView::drop()
 {
+    DatabaseCatalog::instance().removeDependency(select_table_id, getStorageID());
```

Comment:
but it should be dropped in `shutdown()`, which is always called before `drop()`

Reply:
Sorry, I didn't dig into `InterpreterDropQuery` before. I found a bug that might affect both window view and materialized view, I thought it was caused by not removing dependency, but it doesn't. The bug can be reproduced by the following comment.

By the way, `StorageMaterializedView` will remove the dependency in both `shutdown()` and `drop()`, is there any consideration?

Reply:
> By the way, StorageMaterializedView will remove the dependency in both shutdown() and drop(), is there any consideration?

don't know why, may be for no reason, AFAIK `shutdown()` is guaranteed to be called before `drop()`, but ok let's remove dependency two times as it is done for mv.

Reply:
I found that moving addDependency from constructor to `startup()` will fix https://github.com/ClickHouse/ClickHouse/issues/37237.